oauth2_http/java/com/google/auth/oauth2/IdTokenCredentials.java

```diff
@@ -0,0 +1,229 @@
+ *     .setTargetAudience(targetAudience).build();
+ * 
+ * // for ServiceAccountCredentials
+ * ServiceAccountCredentials saCreds = ServiceAccountCredentials.fromStream(new FileInputStream(credPath));
+ * saCreds = (ServiceAccountCredentials) saCreds.createScoped(Arrays.asList("https://www.googleapis.com/auth/iam"));
+ * IdTokenCredentials tokenCredential = IdTokenCredentials.create(saCreds, targetAudience);
+ *
+ * // for ComputeEngineCredentials
+ * ComputeEngineCredentials caCreds = ComputeEngineCredentials.create();
+ * IdTokenCredentials tokenCredential = IdTokenCredentials.create(caCreds, targetAudience,
+ *     Arrays.asList(ComputeEngineCredentials.ID_TOKEN_FORMAT_FULL));
+ *
+ * // for ImpersonatedCredentials
+ * ImpersonatedCredentials imCreds = ImpersonatedCredentials.create(saCreds,
+ *     "impersonated-account@project.iam.gserviceaccount.com", null,
+ *     Arrays.asList("https://www.googleapis.com/auth/cloud-platform"), 300);
+ * IdTokenCredentials tokenCredential = IdTokenCredentials.create(imCreds, targetAudience,
+ *     Arrays.asList(ImpersonatedCredentials.INCLUDE_EMAIL));
+ *
+ * // Use the IdTokenCredential in an authorized transport
+ * GenericUrl genericUrl = new GenericUrl("https://example.com");
+ * HttpCredentialsAdapter adapter = new HttpCredentialsAdapter(tokenCredential);
+ * HttpTransport transport = new NetHttpTransport();
+ * HttpRequest request = transport.createRequestFactory(adapter).buildGetRequest(genericUrl);
+ * HttpResponse response = request.execute();
+ *
+ * // Print the token, expiration and the audience
+ * System.out.println(tokenCredential.getIdToken().getTokenValue());
+ * System.out.println(tokenCredential.getIdToken().getJsonWebSignature().getPayload().getAudienceAsList());
+ * System.out.println(tokenCredential.getIdToken().getJsonWebSignature().getPayload().getExpirationTimeSeconds());
+ * </pre>
+ */
+@Beta
+public class IdTokenCredentials extends OAuth2Credentials {
+
+  private static final long serialVersionUID = -2133257318957588431L;
+
+  private IdTokenProvider idTokenProvider;
+  private String targetAudience;
+  private List<IdTokenProvider.Option> options;
+
+
+  /**
+   * Returns IdToken credentials associated with the sourceCredentials and with an audience
+   * specified. Specify extensions and additional claims for the IdToken by applying any approprite
+   * Options for the given credential type.
+   *
+   * @param idTokenProvider The source credential for the Id Token that implements IdTokenProvider 
+   * @param targetAudience The audience field for the issued ID Token
+   * @param options List of Credential specific options for for the token. For example, an IDToken
+   *     for a ComputeEngineCredential can return platform specific claims if
+   *     "ComputeEngineCredentials.ID_TOKEN_FORMAT_FULL" is provided as a list option.
+   *     tokens.
+   * @return IdTokenCredential
+   */
+  public static IdTokenCredentials create(
```

Comment:
I don't think we should provide this (or the version without options) when the builder is available. It's not adding any convenience, is extra surface area, and could be confusing about which one to use.

If we are confident the interface will never be expanded, we could prefer the static create over the builder, but the builder is more flexible for the future.